core/src/tile/tileManager.cpp

```diff
@@ -352,31 +351,36 @@ void TileManager::enqueueTask(TileSet& _tileSet, const TileID& _tileID,
 // create and download raster references store these
 // rastertasks in this datasource' task
 void TileManager::loadSubTasks(std::vector<std::shared_ptr<DataSource>>& _subSources,
-                                  std::shared_ptr<TileTask>& tileTask, const TileID& tileID) {
+                               std::shared_ptr<TileTask>& tileTask, const TileID& tileID) {
+
     auto& subTasks = tileTask->subTasks();
 
     if (subTasks.size() >= _subSources.size()) { return; }
 
-    // FIXME tasks must be in the correct order to get added to the right
-    // texture slot!
-
-    for (size_t index = subTasks.size(); index < _subSources.size(); index++) {
+    for (size_t index = 0; index < _subSources.size(); index++) {
         auto& subSource = _subSources[index];
+        int subTaskId = index + 1;
+
+        auto it = std::lower_bound(subTasks.begin(), subTasks.end(), subTaskId,
+                                   [&](auto& task, auto id){ return task->subTaskId() < id; });
+
+        if (it != subTasks.end() && (*it)->subTaskId() == subTaskId) { continue; }
+
         TileID subTileID = tileID;
         if (subTileID.z > subSource->maxZoom()) {
             subTileID = subTileID.withMaxSourceZoom(subSource->maxZoom());
         }
-        auto subTask = subSource->createTask(subTileID, true);
+        auto subTask = subSource->createTask(subTileID, index+1);
```

Comment:
Yup... the the change in https://github.com/tangrams/tangram-es/pull/717/commits/08d627357e333889b11b7a903a0581c8aab29476